### PR TITLE
uv_link_t: initial

### PR DESCRIPTION
**Pull request by indutny**
- [x] image
- [x] discuss error reporting

### CODE REVIEW

**Comment by indutny** [comment]
I should probably expand this with some fancy illustrations of how it all works internally.

**Comment by indutny** [comment]
Hooray, added image.

**Comment by indutny** [comment]
No reviews :( Landing as it is in 9a589a9.

File: src/e.uv-link-t.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+to the users, since [uv_link_t][9] is intended to be used together with
+[libuv][0].
+
+Here is a visual explanation of how [uv_link_t][0] works:
+
+![uv_link_source_t][17]
+
+## Examples
+
+Before we will take a peek look at the APIs, let's discuss what can be done with
```
Comment (trevnorris) [inline src/e.uv-link-t.md:61]: "Before we take a peek at the APIs"

File: src/e.uv-link-t.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+[uv_link_t][9]. Technically, any stream-based (TCP-based) protocol can be
```
Comment (trevnorris) [inline src/e.uv-link-t.md:62]: would it be accurate to say "any stream-based (i.e. anything that uses `uv_stream_t`)"?

File: src/e.uv-link-t.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+vim project.gyp
+```
+
+```json
+{
+  "variables": {
+    "gypkg_deps": [
+      "git://github.com/libuv/libuv.git@^1.9.0 => uv.gyp:libuv",
+      "git://github.com/indutny/uv_link_t@^1.0.0 [gpg] => uv_link_t.gyp:uv_link_t",
+    }
```
Comment (trevnorris) [inline src/e.uv-link-t.md:197]: because you're calling this `json` it seems unhappy w/ the trailing comma.

**Comment by trevnorris** [comment]
Sorry for the late review! Just left a few minor comments, but otherwise looks like a great post.

File: src/e.uv-link-t.md (hunk 1 above)
Comment (indutny) [inline src/e.uv-link-t.md:61]: Ack.

File: src/e.uv-link-t.md (hunk 2 above)
Comment (indutny) [inline src/e.uv-link-t.md:62]: Ack.

File: src/e.uv-link-t.md (hunk 3 above)
Comment (indutny) [inline src/e.uv-link-t.md:197]: Right, let's call it python :)

**Comment by indutny** [comment]
@trevnorris no worries, pushed fixes to the blog!

**Pull request closed without merging**